src/primval.rs

```diff
@@ -14,6 +14,8 @@ pub enum PrimVal {
 
     Ptr(Pointer),
     FnPtr(Pointer),
+    VtablePtr(Pointer, Pointer),
+    SlicePtr(Pointer, u64),
```

Comment:
It probably wouldn't be too hard to switch from this version to a `ByValPair` version.